File: src/main/java/leetcode/leet2.java
package leetcode;


import leetclass.ListNode;

public class leet2 {
    public static void main(String[] args) {
      int  a[]={2,4,3};
      int  b[]={5,6,4};
      //定义a为ListNode
      ListNode l1=ListNode.makeNode(a);
      ListNode l2=ListNode.makeNode(b);

      ListNode  s=addTwoNumbers(l1,l2);
      System.out.println(ListNode.traverse(s));

    }

    public static ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode pre = new ListNode(0);
        ListNode cur = pre;
        int carry = 0;
        while(l1 != null || l2 != null) {
            int x = l1 == null ? 0 : l1.val;
            int y = l2 == null ? 0 : l2.val;
            int sum = x + y + carry;

            carry = sum / 10;
            sum = sum % 10;
            cur.next = new ListNode(sum);

            cur = cur.next;
            if(l1 != null)
                l1 = l1.next;
            if(l2 != null)
                l2 = l2.next;
        }
        if(carry == 1) {
            cur.next = new ListNode(carry);
        }
        return pre.next;


    }

}

File: python/python/leetcode/leet2.py
from typing import Optional

from leetclass.ListCode import make_node, node_list, ListNode


class Solution:
    @staticmethod
    def add_two_numbers(l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        pre=ListNode(0)
        cur=pre
        carry=0
        while l1 is not None or l2 is not None:
                if l1 is not None:
                    x=l1.val
                else:
                    x=0
                if l2 is not None:
                    y=l2.val
                else:
                    y=0
                val=int(x+y+carry)
                carry=int(val/10)
                val=int(val%10)
                cur.next=ListNode(val)
                cur=cur.next
                if l1 is not None:
                    l1=l1.next
                if l2 is not None:
                    l2=l2.next
        if carry==1:
          cur.next=ListNode(carry)
        return pre.next



if __name__ == '__main__':
    l1 = [2, 4, 3]
    l2 = [5, 6, 4]
    node_l1 = make_node(l1)
    node_l2 = make_node(l2)
    list_node = Solution.add_two_numbers(node_l1, node_l2)
    print(node_list(list_node))
